core/iam_authenticator.go

```diff
@@ -32,12 +32,17 @@ const (
 	REQUEST_TOKEN_RESPONSE_TYPE = "cloud_iam"
 )
 
-// IamAuthenticator : This authenticator will automatically fetch an access token for the
-// configured apikey.  Outbound REST requests invoked by the BaseService are then authenticated
-// by adding a Bearer-type Authorization header containing the access token.
+// IamAuthenticator uses an apikey to obtain a suitable bearer token value,
+// and adds the bearer token to requests via an Authorization header
+// of the form:
+// pair to obtain a suitable bearer token, and adds it to requests.
+//
+// The bearer token will be sent as an Authorization header in the form:
+//
+// 		Authorization: Bearer <bearer-token>
```

Comment:
This looks like a mix of the old and new text -- not clear to me what should stay and what should be removed, but some change is needed.